src/main/java/com/hierynomus/smbj/session/Session.java

```diff
@@ -93,6 +103,9 @@ private Share connectTree(final String shareName) {
         SmbPath smbPath = new SmbPath(remoteHostname, shareName);
         logger.info("Connection to {} on session {}", smbPath, sessionId);
         try {
+            if (connection.getConfig().isDFSEnabled()) {
```

Comment:
The main question is who owns the whole of the DFS resolver, which is basically the set of data structures described in MS-DFSC (mainly, the DomainCache and ReferralCache).  Right now, these are all in a singleton in DFS.java.  That DFS object _could_ be owned by the SMBClient, but I wasn't completely sure about it, since I don't have a use case for that.